src/akar/util.clj

```diff
@@ -0,0 +1,48 @@
+(ns akar.util
+  (:require [clojure.walk :as cw]))
+
+
+(declare ^{:doc "All the occurances of this symbol will be replaced by
+recursive function call when used inside body passed to
+defn-trampolined.  DO NOT use this symbol outside of
+defn-trampolined."} trampolined-recur)
+
+
+(defn ^:private postwalk-fn-replace
+  "Calls the matching-fn on every expression and if matching-fn
+  returned true, replaces the expression by the result of calling
+  replacing-fn with the given expr as the parameter"
+  [matching-fn replacing-fn form]
+  (cw/postwalk (fn [x]
+                 (if (matching-fn x)
+                   (replacing-fn x)
+                   x))
+               form))
+
+
+(defmacro defn-trampolined
+   "Given a function body which uses trampolined-recur for
+  recursion (instead of recur), this macro generates a function with
```

Comment:
I am not sure about the indentation convention for docstrings on macros and functions. 

I usually refer to Prisma folks' code when in doubt. They indent their docstring with two spaces for the first line, and then three spaces for following lines, so the characters align with each other instead of with the quotation mark. Could you please fix that?

Reply:
Thanks for the suggestion. I have used the formatting you have suggested.